typings/fengari.d.ts

```diff
@@ -0,0 +1,186 @@
+    function lua_isnumber(L: lua_State, index: number): boolean;
+    function lua_isproxy(L: lua_State, index: number): boolean;
+    function lua_isstring(L: lua_State, index: number): boolean;
+    function lua_istable(L: lua_State, index: number): boolean;
+    function lua_isuserdata(L: lua_State, index: number): boolean;
+    function lua_isthread(L: lua_State, index: number): boolean;
+    function lua_isyieldable(L: lua_State, index: number): boolean;
+
+    function lua_len(L: lua_State, index: number): number;
+    function lua_load(L: lua_State, reader: lua_Reader, data: any, chunkname: lua_String, mode: lua_String | null): number;
+
+    function lua_newstate(f: lua_Alloc, ud: any): lua_State;
+    function lua_newtable(L: lua_State): void;
+    function lua_newthread(L: lua_State): lua_State;
+
+    function lua_pcall(L: lua_State, nargs: number, nresults: number, errfunc: number): void;
+    function lua_pop(L: lua_State, n: number): void;
+    function lua_pushboolean(L: lua_State, b: boolean): void;
+    function lua_pushnumber(L: lua_State, n: number): void;
+    function lua_pushinteger(L: lua_State, n: number): void;
+    function lua_pushnil(L: lua_State): void;
+    function lua_pushliteral(L: lua_State, s: string): void;
+    function lua_pushstring(L: lua_State, s: lua_String | string): void;
+    function lua_pushlstring(L: lua_State, s: lua_String | string, size: number): void;
+    function lua_pushvalue(L: lua_State, index: number): void;
+    function lua_pushjsfunction(L: lua_State, fn: lua_Cfunction): void;
+    function lua_pushcfunction(L: lua_State, fn: lua_Cfunction): void;
+    function lua_pushjsclosure(L: lua_State, fn: lua_Cfunction, idx: number): void;
+    function lua_pushcclosure(L: lua_State, fn: lua_Cfunction, idx: number): void;
+
+    function lua_rawget(L: lua_State, index: number): void;
+    function lua_rawgeti(L: lua_State, index: number, n: number): void;
+    function lua_rawset(L: lua_State, index: number): void;
+    function lua_rawseti(L: lua_State, index: number, n: number): void;
+
+    function lua_settable(L: lua_State, index: number): void;
+    function lua_setfield(L: lua_State, index: number, s: lua_String): void;
+    function lua_setglobal(L: lua_State, s: lua_String): void;
+
+    function lua_toboolean(L: lua_State, index: number): boolean;
+    function lua_tointeger(L: lua_State, index: number): lua_Integer;
+    function lua_tonumber(L: lua_State, index: number): lua_Number;
+    function lua_tostring(L: lua_State, index: number): lua_String;
+    function lua_tojsstring(L: lua_State, index: number): string;
+    function lua_tovalue(L: lua_State, index: number): string;
+    function lua_touserdata(L: lua_State, index: number): any;
+
+    function lua_upvalueindex(index: number): number;
+  }
+
+  export namespace lualib {
+    function luaL_openlibs(L: lua.lua_State): void;
+  }
+
+  export namespace lauxlib {
+    // tslint:disable:class-name
```

Comment:
Can we just disable this for the whole file?

Reply:
will do.